ha/monitor/k8s/monitor.py

```diff
@@ -15,35 +15,104 @@
+        producer = self._get_producer()
+
+        # Change to multiprocessing
+        # Creating NODE monitor object
+        self.node_monitor = ObjectMonitor(producer, K8SClientConst.NODE, **kwargs)
+
+        pod_labels = Conf.get(const.HA_GLOBAL_INDEX, "data_pod_label")
+        pod_label_str = ', '.join(pod_label for pod_label in pod_labels)
+        # TODO : Change 'name' field to 'app' in label_selector if required.
+        kwargs[K8SClientConst.LABEL_SELECTOR] = f'name in ({pod_label_str})'
+
+        # Creating POD monitor object
+        self.pod_monitor = ObjectMonitor(producer, K8SClientConst.POD, **kwargs)
+
+    def _get_producer(self):
+        """
+        Get message bus producer
+        """
+        message_type = Conf.get(const.HA_GLOBAL_INDEX, f"MONITOR{_DELIM}message_type")
+        producer_id = Conf.get(const.HA_GLOBAL_INDEX, f"MONITOR{_DELIM}producer_id")
+        Log.info("Initializing message bus.")
+        MessageBus.init()
+        Log.info(f"Getting producer {producer_id} for message type: {message_type}")
+        return MessageBus.get_producer(producer_id, message_type)
+
+    def set_sigterm(self, signum, frame):
+        """
+        Callback function to receive a signal
+        """
+        Log.info(f"Received SIGTERM {signum}.")
+        Log.debug(f"Received signal: {signum} during execution of frame: {frame}")
+        Log.info("Stopping the K8s Monitor...")
+        self.node_monitor.set_sigterm(signum, frame)
+        self.pod_monitor.set_sigterm(signum, frame)
+
+    def start(self):
+        """
+        start the threads
+        """
+        Log.info("Starting all threads of k8s Monitor...")
+        self.node_monitor.start()
+        self.pod_monitor.start()
+        Log.info(f"K8s Monitor with PID {os.getpid()} started successfully.")
+
+    def wait_for_exit(self):
+        """
+        join and wait for all threads to exit
+        """
+        self.node_monitor.join()
+        self.pod_monitor.join()
+        Log.info(f"K8s Monitor with PID {os.getpid()} stopped successfully.")
+
 
 if __name__ == "__main__":
-    object_threads = []
-    # Read I/O pod selector label from ha.conf . Will be received from provisioner confstore
-    # provisioner needs to be informed to add it in confstore  (to be added there )
-    ConfigManager.init("k8s_monitor")
-    pod_labels = Conf.get(const.HA_GLOBAL_INDEX, "data_pod_label")
-    pod_label_str = ', '.join(pod_label for pod_label in pod_labels)
-
-    kwargs = {'pretty': True}
-    # Change to multiprocessing
-    node_thread = ObjectMonitor('node', **kwargs)
-    # TODO : Change 'name' field to 'app' in label_selector if required.
-    kwargs['label_selector'] = f'name in ({pod_label_str})'
-    pod_thread = ObjectMonitor('pod', **kwargs)
-
-    pod_thread.start()
-    node_thread.start()
-
-    object_threads.append(pod_thread)
-    object_threads.append(node_thread)
-
-    for a_thread in object_threads:
```

Comment:
minor comment. 
See if we can have a list of object threads like here.
That way when monitor gets extended to have watchers for containers , PVCs , it will be simpler to itherate over list and join all threads ,

Reply:
done